test/tcp_connection_test.exs

```diff
@@ -2,45 +2,43 @@ defmodule SonicClient.TcpConnectionTest do
   use ExUnit.Case
   alias SonicClient.TcpConnection
 
-  describe "#start_link" do
+  describe "#open" do
     test "Successful connection" do
-      assert {:ok, conn} = TcpConnection.start_link(host(), 1491, [])
-      assert {:ok, message} = TcpConnection.recv(conn)
-      assert "CONNECTED" <> _ = message
+      {:ok, conn} = TcpConnection.open(host(), 1491)
       TcpConnection.close(conn)
     end
   end
 
-  describe "#send" do
+  describe "#request" do
     test "send start search message" do
-      {:ok, conn} = TcpConnection.start_link(host(), 1491, [])
-      assert :ok = TcpConnection.send(conn, "START search SecretPassword")
+      {:ok, conn} = TcpConnection.open(host(), 1491)
+      assert {:ok, _msq} = TcpConnection.request(conn, "START search SecretPassword")
```

Comment:
Small type in `_msq`, actually can we check the message as well?

Reply:
Hmmm, yep!